### PR TITLE
A lot of stuff happened here

### LINKED ISSUE #13: Mark user as dead (+ delete all his/her locations) in DB upon error code 520
error code 520 means "usero non existo" -> we can safely delete the user and all his/her locations from the DB in this case

### LINKED ISSUE #15: Make debugging great again
We cannot debug the backend right now. This seems to be connected to eventlet and WebSockets

### LINKED ISSUE #18: Introduce logging
We have too many print statements in the server

### LINKED ISSUE #25: Replace WS with REST
We only want to load new data after we explicitly requested it.
WS is overkill because we don't get live data as quickly, just every 30mins.
We can still reload new incoming data with REST

### LINKED ISSUE #27: Minimize data transfer to FE
Latest location endpoint:
Cut lat/lng after 3 decimal places, don't send timestamp

### LINKED ISSUE #36: Endpoint for app activity of users
lastOnlineTime

### PR DESCRIPTION
**Pull request by muellerjonas**
@li1  @FelixSchwarzmeier  new REST endpoints for you (especially for user stats):

'/stats/users_with_most_location_data' (this is slow because of the DB)
'/stats/number_of_locations'
'/stats/number_of_users'

@Christoph-Maximilian Note the introduction of logging. print(...) is forbidden now!

closes #13:
Mark users dead + delete all their locations if Lovoo says they are dead 

closes #15:
Return 100 locations instead of just 50 

closes #25:
Kill WS and replace it with REST

closes #27:
Throw out time in latest_location endpoint 

closes #36:
User stats: Top 100 users regarding amount of location data 

closes #18:
Introduced logging everywhere

### CODE REVIEW

File: db_connector.py [hunk 1]
```diff
@@ -196,13 +200,27 @@ def get_age_distribution(self, split_genders=False):
         else:
             return result
 
+    def get_100_users_with_most_location_data(self):
+        return r.table('locations').group("user_id").count().ungroup().order_by(r.desc("reduction")).limit(100). \
```
Comment (FelixSchwarzmeier) [inline db_connector.py]: Why is there a backslash?^^

File: app/__init__.py [hunk 2]
```diff
@@ -3,6 +3,9 @@
 from flask_socketio import SocketIO  # install with conda install flask-socketio
```
Comment (FelixSchwarzmeier) [inline app/__init__.py]: Please remove Socket-IO from BE

File: app/__init__.py [hunk 3]
```diff
@@ -21,4 +24,4 @@
 app.config['CORS_HEADERS'] = 'Content-Type'
 ###app.config['SECRET_KEY'] = 'secret!'
 socketio = SocketIO(app, async_mode=('eventlet' if use_eventlet else 'threading'), use_reloader=False)  # this is Flask WebSockets
-print("Created app and socketio with " + ('eventlet' if use_eventlet else 'threading'))
+logging.info("Created app and socketio with " + ('eventlet' if use_eventlet else 'threading'))
```
Comment (FelixSchwarzmeier) [inline app/__init__.py]: No more socket-IO!

File: db_connector.py (hunk 1 above)
Comment (muellerjonas) [reply db_connector.py]: Because Python.
There's no curly braces so multiple lines that should be treated as a single line are separated by a \

Comment (FelixSchwarzmeier) [reply db_connector.py]: Ah good to know👍

**Comment by muellerjonas** [comment]
Forgot to mention: This also closes #23

**Comment by muellerjonas** [comment]
Also closes #38

**Pull request merged**